File: C++/src/Solution401_500/Solution412.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution412.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<string> Solution412::fizzBuzz(int n) {
    vector<string> res;
    for (int i = 1; i <= n; ++i) {
        if (i % 3 == 0 && i % 5 == 0){
            res.push_back("FizzBuzz");
        }else if (i % 3 == 0){
            res.push_back("Fizz");
        }else if (i % 5 == 0){
            res.push_back("Buzz");
        } else{
            res.push_back(to_string(i));
        }
    }
    return res;
}

File: Java/src/main/java/QuestionBank/Solution401_500/Solution412.java
package QuestionBank.Solution401_500;

import java.util.ArrayList;
import java.util.List;

/**
 * @author windows
 */
public class Solution412 {
    public List<String> fizzBuzz(int n) {
        ArrayList<String> list = new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            if (i % 3 == 0 && i % 5 == 0){
                list.add("FizzBuzz");
            }else if (i % 3 == 0){
                list.add("Fizz");
            }else if (i % 5 == 0){
                list.add("Buzz");
            }else{
                list.add(String.valueOf(i));
            }
        }
        return list;
    }


    public static void main(String[] args) {
        Solution412 solution412 = new Solution412();
    }
}